src/charm.pyi

```diff
@@ -26,9 +27,33 @@ class ThemeInfoDict(TypedDict):
     version: str
 
 
+class AddonInfoDict(TypedDict):
+    name: str
+    status: str
+    update: str
+    version: str
+
+
+class WPAddonListExecResult(NamedTuple):
+    success: bool
+    result: Optional[List[AddonInfoDict]]
+    message: str
+
+
+class NoneReturnExecResult(NamedTuple):
+    success: bool
+    result: None
+    message: str
+
+
 class WordpressCharm(ops.charm.CharmBase):
     class _ReplicaRelationNotReady(Exception): ...
 
+    class _ExecResult(NamedTuple):
+        success: bool
+        result: Any
```

Comment:
Is the result variable?

Reply:
Yes, I will add proper typing after generic named tuple is fully supported in the future.